src/box/lua/net_box.c

```diff
@@ -1873,12 +1880,15 @@ netbox_iproto_auth(struct lua_State *L)
 	int fd = lua_tointeger(L, 5);
 	struct ibuf *send_buf = (struct ibuf *)lua_topointer(L, 6);
 	struct ibuf *recv_buf = (struct ibuf *)lua_topointer(L, 7);
+	struct fiber_cond *send_cond =
+		(struct fiber_cond *)luaL_checkudata(L, 8, cond_typename);
```

Comment:
`netbox_iproto_auth` requires condvar signaling, as well as console, to keep the close () code simpler. Otherwise in `close`, we must check that we have state == 'active' or state == 'fetch_schema', as well as that we are not using the console (otherwise, upon shutdown, we risk hang).

Reply:
Makes sense